samples/export.py

```diff
@@ -0,0 +1,69 @@
+import argparse
+import getpass
+import logging
+
+import tableauserverclient as TSC
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Get all of the refresh tasks available on a server')
+    parser.add_argument('--server', '-s', required=True, help='server address')
+    parser.add_argument('--username', '-u', required=True, help='username to sign into server')
+    parser.add_argument('--site', '-S', default=None)
+    parser.add_argument('-p', default=None)
+
+    parser.add_argument('--logging-level', '-l', choices=['debug', 'info', 'error'], default='error',
+                        help='desired logging level (set to error by default)')
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('--pdf', dest='type', action='store_const', const=('populate_pdf', 'PDFRequestOptions', 'pdf',
+                                                                          'pdf'))
+    group.add_argument('--png', dest='type', action='store_const', const=('populate_image', 'ImageRequestOptions',
+                                                                          'image', 'png'))
+    group.add_argument('--csv', dest='type', action='store_const', const=('populate_csv', 'CSVRequestOptions', 'csv',
+                                                                          'csv'))
+
+    parser.add_argument('--file', '-f', help='filename to store the exported data')
+    parser.add_argument('--filter', '-vf', metavar='COLUMN:VALUE',
+                        help='View filter to apply to the view')
+    parser.add_argument('resource_id', help='LUID for the view')
+
+    args = parser.parse_args()
+
+    if args.p is None:
+        password = getpass.getpass("Password: ")
+    else:
+        password = args.p
+
+    # Set logging level based on user input, or error by default
+    logging_level = getattr(logging, args.logging_level.upper())
+    logging.basicConfig(level=logging_level)
+
+    tableau_auth = TSC.TableauAuth(args.username, password, args.site)
+    server = TSC.Server(args.server, use_server_version=True)
+    with server.auth.sign_in(tableau_auth):
+        views = filter(lambda x: x.id == args.resource_id,
+                       TSC.Pager(server.views.get))
+        view = views.pop()
+        (populate_name, option_factory_name, member_name, extension) = args.type
```

Comment:
It took me a few reads to understand what's going on here. 
I think the code is fine, but some suggestions for clarity:

- Rename 'populate_name' to 'populate_func_name'
- Add a one line comment above that says something like "We've got lots of types and classes, this is code to figure out all the right functions to call" or something